src/DiffResults_helpers.jl

```diff
@@ -25,5 +25,6 @@ constructed with [`diffresults_buffer`](@ref). Gradient is not copied as caller
 vector.
 """
 function _diffresults_extract(diffresult::DiffResults.DiffResult)
+    # NOTE: Is this still needed?
```

Comment:
That's not affected by the change above, is it?

Reply:
I also don't understand why it would not be needed.

Reply:
Nah, it was a note by me so we I could ask why it's there:) As in, can we just remove it?

Reply:
What would you use instead?

The purpose was to collect everything related to extraction in a single function. I am open to suggestions, but simply removing it would break the package (the function is used).

Reply:
Ah, I thought the `::Real` was the reason for it, and was thinking maybe this is now redundant because type-inference is better / DiffResults might have fixed whatever issue required the `::Real` in the first place. If that's not the case and it's just for convenience, then I'll remove the comment:)

Reply:
I no longer recall the case for `::Real`, so I guess you can remove it. We can always put it back if things break and then document why it is there.

Please also remove the comment, and then I am happy to merge.

Reply:
Done:)